src/main/scala/io/moia/scalaHttpClient/HttpClient.scala

```diff
@@ -132,16 +133,15 @@ abstract class HttpLayer[LoggingContext](
     response.toStrict(retryConfig.strictifyResponseTimeout)
   }
 
-  private[this] def shouldBeRetried(statusCode: StatusCode): Boolean =
-    Seq(StatusCodes.RequestTimeout, StatusCodes.TooManyRequests, StatusCodes.ServiceUnavailable, StatusCodes.InternalServerError)
-      .contains(statusCode)
-
   private[this] def retryCount(statusCode: StatusCode): Int = statusCode match {
     case StatusCodes.RequestTimeout      => retryConfig.retriesRequestTimeout
     case StatusCodes.TooManyRequests     => retryConfig.retriesTooManyRequests
+    case StatusCodes.ClientError(_)      => retryConfig.retriesClientError
+    case StatusCodes.InternalServerError => retryConfig.retriesInternalServerError
+    case StatusCodes.BadGateway          => retryConfig.retriesBadGateway
     case StatusCodes.ServiceUnavailable  => retryConfig.retriesServiceUnavailable
-    case StatusCodes.InternalServerError => retryConfig.retriesServerError
-    case _                               => retryConfig.retriesServerError
+    case StatusCodes.ServerError(_)      => retryConfig.retriesServerError
+    case _                               => 0 // default: don't retry weird/custom codes
```

Comment:
Hmm maybe logging the code would be help diagnosing why a certain request was not retried?